boundary_checker_impr.py

```diff
@@ -0,0 +1,82 @@
+"""Boundary checking (improved)
```

Comment:
This file seems to be a duplicate of example/processors/boundary_checker_impr.py - I think perhaps it should be removed?